pkg/core/interop/context.go

```diff
@@ -63,13 +63,15 @@ type Context struct {
 	VM               *vm.VM
 	Functions        []Function
 	Invocations      map[util.Uint160]int
+	InvocationCalls  []state.ContractInvocation
```

Comment:
s/InvocationCalls/InvocationTree

> It was unclear to me based on https://github.com/nspcc-dev/neo-go/pull/3569#discussion_r1746103762 if I should have made it a tree or keep it flat.

Regarding this: if it's not hard, then let's make it a proper tree. It may be beneficial for some situations. AFAIU, Roman agrees with it.